### PR TITLE
Centralize project manage check in Project class

### PR DESCRIPTION
**Pull request by cpeel**
We currently have two similar checks to determine if a user is able to manage a project. Centralize on the one within the Project class. This ensures that the API (which uses the Project-based method for checking wordlist management access) and the project edit UI (which uses the `project_edit.inc`-based way) use the same code.

Testable in the [single-manage-check](https://www.pgdp.org/~cpeel/c.branch/single-manage-check/) sandbox.

### CODE REVIEW

File: tools/project_manager/editproject.php [hunk 1]
```diff
@@ -288,15 +278,9 @@ public function set_from_post()
             $projectid = get_projectID_param($_POST, 'projectid');
             $this->projectid = $projectid;
 
-            $ucep_result = user_can_edit_project($this->projectid);
-            if ($ucep_result == PROJECT_DOES_NOT_EXIST) {
-                return _("parameter 'projectid' is invalid: no such project").": '$this->projectid'";
-            } elseif ($ucep_result == USER_CANNOT_EDIT_PROJECT) {
+            $project = new Project($this->projectid);
```
Comment (chrismiceli) [inline tools/project_manager/editproject.php:281]: do we want to catch here and return the not found message like before with PROJECT_DOES_NOT_EXIT return response?

**Review by chrismiceli — APPROVED** [review]

File: tools/project_manager/editproject.php (hunk 1 above)
Comment (cpeel) [reply tools/project_manager/editproject.php:281]: Yes, good catch!

**Review by chrismiceli — APPROVED** [review]

**Comment by 70ray** [comment]
This has not been changed by this but in some cases the message says "You are not allowed ... " and in other cases "You are not authorized ... ".
Would it be a good idea to return a code and a message like can_be_proofed_by_current_user() does then the message could be defined in one place?

**Comment by cpeel** [comment]
> This has not been changed by this but in some cases the message says "You are not allowed ... " and in other cases "You are not authorized ... ".
> Would it be a good idea to return a code and a message like can_be_proofed_by_current_user() does then the message could be defined in one place?

Good eye - I hadn't noticed that. Perhaps if we were creating this function from scratch again I would make this more like `can_be_proofed_by_current_user()`, but the existing function is used all over the place and a string isn't always needed. That would turn something that looks like this:
```php
if ($project->can_be_managed_by_user) {
}
```
into
```php
$message = $project->can_be_managed_by_user
if (!$message) {
}
```
which is less intuitive.

I will go back as part of this MR and do a cleanup to make the messages consistent.

**Comment by cpeel** [comment]
I've standardized the project access error messages across the site, as well as the error messages in `tools/site_admin/*`. We have a few other "You are not ..." messages but they are all more specific and don't lend themselves to be one of these generic options.

**Review by 70ray — APPROVED** [review]

**Review by srjfoo — COMMENTED** [review]
I haven't quite finished testing, but it's time for me to stop for the night, and I wanted to comment on what I've done so far. In addition to the other two comments, what I've found when testing the scripts is that I'm not seeing the new messages in the sandbox. I've looked at about 10 of them (working from the bottom towards the top), and except for the ones where the message is the same, and the change was just adding the `gettext()` markup, every message I'm seeing in the `single-manage-check` sandbox is the same as the old message as shown in the main TEST sandbox.

File: tools/project_manager/move_projects.php [hunk 2]
```diff
@@ -35,9 +34,8 @@
         continue;
     }
 
-    $result = user_can_edit_project($projectid);
-    if ($result == USER_CANNOT_EDIT_PROJECT) {
-        echo "    " . _("You are not authorize to manage this project.") . "\n";
+    if (!$project->can_be_managed_by_current_user) {
+        echo "    " . _("You are not authorized to manage this project.") . "\n";
```
[the remark this answers is not in the document]
Comment (cpeel) [reply tools/project_manager/move_projects.php:38]: Shockingly, this file isn't actually wired into the site anywhere. Literally nothing points to this, even in `noncvs`. The file was added in 2003 and the last change of note was made in 2006.

It hasn't been accessed on PROD in the last month (the length of time we keep Apache access logs).

I think we should just delete this file.

**Comment by cpeel** [comment]
> I've looked at about 10 of them (working from the bottom towards the top), and except for the ones where the message is the same, and the change was just adding the `gettext()` markup, every message I'm seeing in the `single-manage-check` sandbox is the same as the old message as shown in the main TEST sandbox.

This is probably developer error. I suspected I didn't refresh the sandbox when I made the second commit. I've done so now and confirmed the error message is there for the site admin pages. Sorry about that!

File: tools/project_manager/move_projects.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (srjfoo) [reply tools/project_manager/move_projects.php:38]: Given that, deleting it sounds like a good idea to me, too.

**Comment by srjfoo** [comment]
There are a couple more things I want to go back through, but I've been through all of the scripts, now, and everything looks good. I did find one that probably ought to be added: `tools/pending_access_requests.php`. It's limited to SAs and Access Request Reviewers, but still has the old "permission denied" message.

**Comment by cpeel** [comment]
> I did find one that probably ought to be added: `tools/pending_access_requests.php`. It's limited to SAs and Access Request Reviewers, but still has the old "permission denied" message.

Thanks for finding that one. I've pushed up a fix and updated the sandbox.

File: tools/project_manager/move_projects.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (srjfoo) [reply tools/project_manager/move_projects.php:38]: Are you planning to delete this file in this MR, or does that need more research and/or a separate MR?

**Review by srjfoo — APPROVED** [review]
There's still a minor inconsistency in the "You are not authorized to manage this project." Some have the form:

> You are not authorized to manage this project`. (`projectID...`).`

Some:

> You are not authorized to manage this project`.:` projectID...

And some show no projectid at all. That's all stuff that can be sorted out later, if ever, though.

File: tools/project_manager/move_projects.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (cpeel) [reply tools/project_manager/move_projects.php:38]: I'm going to do this in a separate MR. Going to go through the entire codebase and make sure we don't have any other stragglers.

**Pull request merged**